kyco/core/tcp_server.py

```diff
@@ -72,36 +72,41 @@ def setup(self):
     def handle(self):
         curr_thread = current_thread()
         header_len = 8
-        while True:
-            # TODO: How to consider the OpenFlow version here?
-            header = Header()
-            binary_data = b''
+        connected = True
+        try:
+            while connected:
+                # TODO: How to consider the OpenFlow version here?
+                header = Header()
+                binary_data = b''
 
-            try:
                 raw_header = self.request.recv(8)
-            except:
-                log.info("Client %s:%s disconnected", self.ip, self.port)
-                break
-
-            while len(raw_header) < header_len:
-                raw_header += self.request.recv(header_len - len(raw_header))
-
-            log.debug("New message from %s:%s at thread %s", self.ip,
-                      self.port, curr_thread.name)
-
-            header.unpack(raw_header)
-
-            message_size = header.length - header_len
-            if message_size > 0:
-                log.debug('Reading the binary_data')
-                binary_data += self.request.recv(message_size)
-
-            # TODO: Do we need other informations from the network packet?
-            content = {'header': header, 'binary_data': binary_data}
-            connection_id = (self.ip, self.port)
-            event = KycoRawOpenFlowMessage(content=content,
-                                           connection_id=connection_id)
-            self.server.controller_put_raw_event(event)
+                if not raw_header:
+                    log.info("Client %s:%s disconnected", self.ip, self.port)
+                    connected = False
+                    break
+
+                while len(raw_header) < header_len:
+                    raw_header += self.request.recv(header_len - len(raw_header))
+
+                log.debug("New message from %s:%s at thread %s", self.ip,
+                          self.port, curr_thread.name)
+
+                header.unpack(raw_header)
+
+                message_size = header.length - header_len
+                if message_size > 0:
+                    log.debug('Reading the binary_data')
+                    binary_data += self.request.recv(message_size)
+
+                # TODO: Do we need other informations from the network packet?
+                content = {'header': header, 'binary_data': binary_data}
+                connection_id = (self.ip, self.port)
+                event = KycoRawOpenFlowMessage(content=content,
+                                               connection_id=connection_id)
+                self.server.controller_put_raw_event(event)
+        except Exception:
```

Comment:
Yes, this exception is quite generic (I think that a commit made after this one has changed the line 107 to a more proper exception).
Anyway, exceptions here would be only on lines 82 and 89.
I think that it would mean a client down, but you have said that it would be, for example, latency timeout.